ocean_lib/ocean/ocean_assets.py

```diff
@@ -666,6 +662,112 @@ def pay_for_service(
 
         return tx_id
 
+    @enforce_types
+    def pay_for_compute_service(
+        self,
+        datasets,
+        algorithm_data,
+        compute_environment,
+        valid_until,
```

Comment:
addressing the first suggestions. However, I'd rather stick with the _data suffix because it is not an asset, it is a set of data describing that asset (as you said, ComputeInput or AlgorithmMetadata)

Reply:
Fair enough.